drivers/gpio/gpio_tca9538.c

```diff
@@ -0,0 +1,601 @@
+	if (!device_is_ready(cfg->i2c_dev)) {
+		LOG_ERR("I2C device not found");
+		return -EINVAL;
+	}
+
+	if ((drv_data->interrupt_enabled) && (rc == 0)) {
+		drv_data->dev = dev;
+		drv_data->gpio_int = device_get_binding(cfg->gpio_int_dev_name);
+
+		if (!drv_data->gpio_int) {
+			rc = -ENOTSUP;
+		} else {
+			k_work_init(&drv_data->work, gpio_tca9538_work_handler);
+
+			gpio_pin_configure(drv_data->gpio_int, cfg->gpio_pin,
+					   GPIO_INPUT | cfg->gpio_flags);
+			gpio_pin_interrupt_configure(drv_data->gpio_int,
+						     cfg->gpio_pin,
+						     GPIO_INT_EDGE_TO_ACTIVE);
+
+			gpio_init_callback(&drv_data->gpio_cb,
+					   gpio_tca9538_init_cb,
+					   BIT(cfg->gpio_pin));
+
+			gpio_add_callback(drv_data->gpio_int,
+					  &drv_data->gpio_cb);
+
+			drv_data->irq_state = (struct tca9538_irq_state){
+				.interrupt_mask = ALL_PINS,
+			};
+		}
+	}
+
+	if (rc == 0) {
+		tx_dat[0] = TCA9538_OUTPUT_PORT;
+		tx_dat[1] = drv_data->pin_state.output;
+		rc = i2c_write(cfg->i2c_dev, tx_dat, sizeof(tx_dat),
+			       cfg->i2c_addr);
+	}
+
+	if (rc == 0) {
+		tx_dat[0] = TCA9538_CONFIGURATION;
+		tx_dat[1] = drv_data->pin_state.dir;
+		rc = i2c_write(cfg->i2c_dev, tx_dat, sizeof(tx_dat),
+			       cfg->i2c_addr);
+	}
+
+	/* Do an initial read, this also clears the interrupt pin */
+	if (rc == 0) {
+		rc = update_input(dev);
+	}
+
+	if (rc != 0) {
+		LOG_ERR("%s init failed: %d", dev->name, rc);
+	} else {
+		LOG_INF("%s init ok", dev->name);
+	}
+	k_sem_give(&drv_data->lock);
+	return rc;
+}
+
+/**
+ * @brief Configures pins of TCA9538 I/O Port.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param pin Pin number (0 - 7)
+ * @param flags Configuration flags, see gpio.h
+ *
+ * @retval 0 If successful.
+ * @retval Negative value for error code.
+ */
+static int gpio_tca9538_config(const struct device *dev, gpio_pin_t pin,
+			       gpio_flags_t flags)
+{
+	const struct tca9538_config *cfg = dev->config;
+	struct tca9538_drv_data *drv_data = dev->data;
+	struct tca9538_pin_state *pins = &drv_data->pin_state;
+	int rc = 0;
+	bool data_first = false;
+	char tx_dat[2];
+
+	/* Can't do I2C bus operations from an ISR */
+	if (k_is_in_isr()) {
+		return -EWOULDBLOCK;
+	}
+
+	k_sem_take(&drv_data->lock, K_FOREVER);
+
+	/* Zephyr currently defines drive strength support based on
+	 * the behavior and capabilities of the Nordic GPIO
+	 * peripheral: strength defaults to low but can be set high,
+	 * and is controlled independently for output levels.
+	 *
+	 * The TCA9538 supports only high strength, and does not
+	 * support different strengths for different levels.
+	 *
+	 * Until something more general is available reject any
+	 * attempt to set a non-default drive strength.
+	 */
+	if ((flags & (GPIO_DS_ALT_LOW | GPIO_DS_ALT_HIGH)) != 0) {
+		rc = -ENOTSUP;
+		goto out;
+	}
+
+	if ((flags & GPIO_OPEN_DRAIN) != 0 || (flags & GPIO_OPEN_SOURCE) != 0) {
+		/* Open drain not supported */
+		/* Open source not supported */
```

Comment:
Let's move the comment above the `if`, like it's done for the pull up / pull down flags. Or we can remove it completely since the code is self explanatory.